apps/autoscheduler/AutoSchedule.cpp

```diff
@@ -1756,6 +1771,28 @@ struct State {
         internal_assert(!binfile.fail()) << "Failed to write " << feature_file;
     }
 
+    bool exceeds_shared_memory_limit(const ScheduleFeatures& feat, const FunctionDAG::Node* node) {
+        if (!gpu_mode()) {
+            return false;
+        }
+
+        static int64_t limit = get_shared_memory_limit();
+
+        if (limit == 0) {
+            return false;
+        }
+
+        // If the working set is too large on the GPU, shared memory will be
+        // exhausted, so reject any such schedules
+        for (const auto& c : root->children) {
```

Comment:
You can tell if something is a child of root by checking feat.inner_parallelism, so here you could say feat.inner_parallelism > 1 && feat.working_set > limit instead of looping